src/hotspot/share/compiler/compilerOracle.cpp

```diff
@@ -778,25 +779,13 @@ void CompilerOracle::print_parse_error(char* error_msg, char* original_line) {
   print_tip();
 }
 
-class LineCopy : StackObj {
-  const char* _copy;
-public:
-    LineCopy(char* line) {
-      _copy = os::strdup(line, mtInternal);
-    }
-    ~LineCopy() {
-      os::free((void*)_copy);
-    }
-    char* get() {
-      return (char*)_copy;
-    }
-};
-
 void CompilerOracle::parse_from_line(char* line) {
   if (line[0] == '\0') return;
   if (line[0] == '#')  return;
 
-  LineCopy original(line);
+  char* original_line = os::strdup(line, mtInternal);
+  auto g = make_guard([&] { os::free(original_line); });
```

Comment:
I have to admit that this usage is very mysterious.  I know you're going to take it out but using lambdas for something that's really hard to parse doesn't seem like a win.  Does save lines of code though.

Reply:
I think a different approach may be possible, that would have usage like this:
```
char* original_line = os::strdup(line, mtInternal); 
ScopeGuard guard([&] { os::free(original_line); });
```
This would use something like (or maybe directly use) a solution to the type-erased lambda capturing problem a la std::function.

Reply:
Yes, this second usage looks like the typical RAII object that is familiar in the code.